### ISSUE TITLE
Show notifications for GitHub Team Discussions

### ISSUE THREAD

**Issue by tmlayton** [octobox/octobox#660]
Are you experiencing this issue in octobox.io or your own instance?
 - [x] octobox.io (I don’t use Octobox.io so it technically is a lie for me to check this, but I’m fairly certain this issue exists there too after looking through the code)
 - [x] My instance

🐞 **Problem**
Our team is trialing GitHub Team Discussions (e.g. https://github.com/orgs/YOUR_ORG/teams/YOUR_TEAM/discussions) and noticed team discussion notifications do not show up in Octobox, although do appear in https://github.com/notifications.

🎯 **Goal**
For those of us using Octobox and GitHub Team Discussions, we would like discussion notifications to show up in Octobox ❤️

**Comment by tmlayton** [comment]
cc/ @jules2689

**Comment by jules2689** [comment]
hmm, Ive not dived into this area before. @chrisarcand @andrew @tarebyte any insights on how to enable this?

**Comment by BPScott** [comment]
I can't see any notifications from teams when hitting https://api.github.com/notifications. I've got a sinking feeling that GitHub don't expose this yet. I'll poke support.

**Comment by tmlayton** [comment]
I think they are available here https://developer.github.com/v3/teams/discussions/

**Comment by BPScott** [comment]
> Hi there Ben,
>
> Thanks for getting in touch! There isn't currently an API for accessing team discussions. Other users have expressed interest and we've shared your +1 for it internally so they can see.

Well, bums.

**Comment by jules2689** [comment]
I'll escalate from Shopify's side as this impacts our employees too

**Comment by jules2689** [comment]
That said, I'm going to close this issue since the issue is on github's side.

**Comment by tmlayton** [comment]
Thanks @BPScott and @jules2689

**Comment by chrisarcand** [comment]
Yeah this is a bummer. Thanks for finding the cause!

**Comment by shayfrendt** [comment]
> I can't see any notifications from teams when hitting https://api.github.com/notifications. I've got a sinking feeling that GitHub don't expose this yet. I'll poke support.

👋 Hey Octobox friends! I'm on the GitHub team that built and released Team Discussions and just wanted to reach out to confirm ~your sinking feeling~ the behavior you're seeing. 😅 

In digging through some project context, it looks like we intentionally excluded team-related notifications from the REST API v3. Here are a couple snippets of insight into that decision:

>- Returning team notifications would be a breaking change because clients can no longer assume that the repository object exists.
>- There isn't anything in the documentation to suggest that the responses you get from the notifications endpoint might have different list types, and so I suspect that many integrators simply expect a non-null "repository" key for every notification in that response.

That being said, hearing from friendly integrators like yourselves helps highlight the different features that folks expect out of our APIs and can help shape our future work.

If notifications support was added to the [GraphQL API](https://developer.github.com/v4/) in the future, is that something that your project would consider using?

**Comment by chrisarcand** [comment]
@shayfrendt Hi!

> If notifications support was added to the GraphQL API in the future, is that something that your project would consider using?

Yes, I would say people would be interested in it. I've always assumed we'd be moving to the V4 API at some point, it's just been pretty low on the list so far.

**Comment by shayfrendt** [comment]
@chrisarcand Ok cool, thanks for the feedback. 🙇 

While we're waiting on the :octocat:, we could set up a notifications mail order catalogue where we load your team discussion notifications onto floppy disks every week and send them to you in the mail Netflix style. 💾  

Ah hmm...that's probably not a great workaround. 🤓 

I'll bring this thread back to the team as a feature stakeholder data point and stuff. Hopefully one day in the not-so-distant future I'd have the pleasure of returning with Exciting News™. 😁

**Comment by Zimmi48** [comment]
Just started with team discussions and was disappointed to see that it wasn't supported by Octobox yet. However, it was great to discover this issue and how fresh and active it is.

I would suggest to keep it open until it's solved so that it is easier to find it (I went through all the open issues before using the issue search).

@shayfrendt Would it be too much to ask to have an option to receive e-mail notifications specifically for team discussions? It would be the best workaround I can see until Octobox support lands.

**Comment by shayfrendt** [comment]
> Would it be too much to ask to have an option to receive e-mail notifications specifically for team discussions? It would be the best workaround I can see until Octobox support lands.

Agreed on email notifications being the best workaround. 👍 

@Zimmi48 It wouldn't be too much to ask, and the fun part is...they already exist! Check out:

https://help.github.com/articles/about-team-discussions/#notifications-for-team-discussions

**Comment by jules2689** [comment]
@Zimmi48 good point

**Comment by Zimmi48** [comment]
@shayfrendt Thanks but AFAICS GitHub doesn't provide any way for me to get *only* team discussion notifications by e-mail. I don't want to receive e-mail notifications for repositories that I follow as this would represent hundreds of notifications and I don't want to configure very smart filters in my mailbox. That's why I'm using Octobox to manage my GitHub notifications... That said, it seems that e-mail notification configuration has become more fine-grained since the time I disabled GitHub e-mail notification entirely. I've just checked the "Email" box in "Notifications for all repositories, teams, or conversations you’re watching." and unchecked all boxes in "Choose which email updates you receive on conversations you’re participating in or watching" that were talking about pull requests or issues. There's no specific mention of team discussions there so if that's the way to only get e-mail notifications for team discussions, this is pretty convoluted...

**Comment by shayfrendt** [comment]
@Zimmi48 Gah why you gotta find all the secret shortcuts we took to get things out the door faster? jk, great 👀 there...

We don't say much about team discussions in this section huh:

![team-discussions](https://user-images.githubusercontent.com/32535/43487186-e42fd766-94e3-11e8-9c5b-0d344c3dbe10.png)

I've followed a follow-up issue on this topic internally.

I don't wanna steer this thread too far off topic with settings/configuration management idea, but definitely feel free to reach out via our [help form](https://github.com/contact) and mention me/this thread and we can discuss further if you like!

**Comment by BPScott** [comment]
@shayfrendt thanks for providing context and for chasing stuff internally <3

**Comment by DouweM** [comment]
@andrew What do you think about reopening this issue as it hasn't actually been resolved yet? This is still blocked by the lack of support for discussion in the REST API (see above) and GraphQL API (https://github.com/github-community/community/discussions/13056), but I think it'd be good to keep this issue open so users can be aware of this limitation.
